src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleDictionary.cs

```diff
@@ -25,8 +25,12 @@ public static partial class JsonSerializer
                 enumerable = (IEnumerable)jsonPropertyInfo.GetValueAsObject(state.Current.CurrentValue);
                 if (enumerable == null)
                 {
-                    // Write a null object or enumerable.
-                    state.Current.WriteObjectOrArrayStart(ClassType.Dictionary, writer, writeNull: true);
+                    if (!state.Current.JsonPropertyInfo.IgnoreNullValues)
```

Comment:
Was ignoring null values on the "read" side already handled? Basically we shouldn't be setting the property to null.

Reply:
> Basically we shouldn't be setting the property to null.

When does that matter? We are constructing a new instance and `null` is the default state for reference types, so I would expect aren't impacting anything by ignoring or not ignoring `null` in the reader.

Reply:
That is the described behavior for IgnoreNull. Perhaps the constructor may have set a field to a non-null values and doesn't want it trampled. It could also be considered a performance optimization (no setter needs to be called).

Reply:
Will add a test covering this scenario and check.

Reply:
Are you planning to address this separately, @tannergooding?

Reply:
Yes, that should probably be handled in a separate PR at this point.